Refactor branch 2.5, clean code

bcode2 opened the pull request:
<!--
If this is your first time contributing to the project (or it's been a while), please consider reviewing https://github.com/quartz-scheduler/contributing/blob/main/CONTRIBUTING.md
-->


This PR...

Fixes issue #

## Changes
refactor: replace while loop with enhanced for loop
refactor: language migration; simplify maps operations
style: improve visibility change l by L for long literals
refactor: replace addAll() with  parameterized constructor
refactor: set serialVersionUID  fields as static and final
refactor: language migration, remove unnecessary boxing
refactor: language migration, use lambda expression to simplify code and remove unused imports
refactor: remove redundant variable declaration
performance: language migration
refactor: simplify boolean expression

-----------------
## Checklist
- [x] tested locally
- [ ] updated the docs
- [ ] added appropriate test
- [x] signed-off on the DCO referenced in the CONTRIBUTING link below via `git commit -s` on my commits, and submit this code under terms of the Apache 2.0 license and assign copyright to the Quartz project owners
  (If you're not using command-line, you can use a [browser extension](https://github.com/scottrigby/dco-gh-ui) )
-----------------
In submitting this contribution, I agree to the terms of contributing as referred to here: 
https://github.com/quartz-scheduler/contributing/blob/main/CONTRIBUTING.md

melloware reviewed (approved):
great work @bcode2

chrisdennis commented on quartz/src/main/java/org/quartz/impl/jdbcjobstore/StdJDBCDelegate.java:
```diff
@@ -3118,10 +3110,9 @@ protected Properties convertToProperty(Map<?, ?> data) throws IOException {
 
-            Object key = entry.getKey();
-            Object val = (entry.getValue() == null) ? "" : entry.getValue();
+            Object key = ((Map.Entry<?, ?>) value).getKey();
```
Unusual casts here? Needed?

chrisdennis commented on quartz/src/main/java/org/quartz/impl/triggers/DailyTimeIntervalTriggerImpl.java:
```diff
@@ -875,8 +875,6 @@ public Set<Integer> getDaysOfWeek() {
     public void setDaysOfWeek(Set<Integer> daysOfWeek) {
         if(daysOfWeek == null || daysOfWeek.isEmpty())
             throw new IllegalArgumentException("DaysOfWeek set must be a set that contains at least one day.");
-        else if(daysOfWeek.isEmpty())
```
Do we want to fix this by changing the previous condition and modifying the messages?

bcode2 replied:
for me it is redundant both messages. The fact would be to choose which one will be left :). There is a pr opened for this :) https://github.com/quartz-scheduler/quartz/issues/1209

chrisdennis commented on quartz/src/main/java/org/quartz/plugins/xml/XMLSchedulingDataProcessorPlugin.java:
```diff
@@ -214,32 +213,29 @@ public void start(UserTransaction userTransaction) {
-                            .usingJobData(FileScanJob.FILE_NAME, jobFile.getFileName())
-                            .usingJobData(FileScanJob.FILE_SCAN_LISTENER_NAME, JOB_INITIALIZATION_PLUGIN_NAME + '_' + getName())
-                            .build();
+                                .usingJobData(FileScanJob.FILE_NAME, jobFile.getFileName())
```
Not a huge fan of this kind of reformatting in a such a large PR.

bcode2 replied:
Yes  could be right, It is just a matter of preference. Do you want me to revert it?,

chrisdennis replied:
I'll let @jhouserizer make the call.

chrisdennis commented on quartz/src/main/java/org/quartz/simpl/CascadingClassLoadHelper.java:
```diff
@@ -107,9 +106,8 @@ public Class<?> loadClass(String name) throws ClassNotFoundException {
-        while (iter.hasNext()) {
-            loadHelper = iter.next();
+        for (ClassLoadHelper helper : loadHelpers) {
+            loadHelper = helper;
```
Seems cleaner to do this assignment right before the `break`

bcode2 replied:
Umm I dont get it. The assignment is done just before the break

chrisdennis replied:
More explicitly:
```java
for (ClassLoadHelper helper : loadHelpers) {
    try {
        clazz = helper.loadClass(name);
        loadHelper = helper;
        break;
    } catch (Throwable t) {
        throwable = t;
    }
}
```
but given the wider context you could go to:
```java
public Class<?> loadClass(String name) throws ClassNotFoundException {
    if (bestCandidate != null) {
        try {
            return bestCandidate.loadClass(name);
        } catch (Throwable t) {
            bestCandidate = null;
        }
    }

    Throwable throwable = null;

    for (ClassLoadHelper helper : loadHelpers) {
        try {
            Class<?> clazz = helper.loadClass(name);
            bestCandidate = helper;
            return clazz;
        } catch (Throwable t) {
            throwable = t;
        }
    }

    if (throwable instanceof ClassNotFoundException) {
        throw (ClassNotFoundException) throwable;
    } else {
        throw new ClassNotFoundException( String.format( "Unable to load class %s by any known loaders.", name), throwable);
    } 
}
```

chrisdennis commented on quartz/src/main/java/org/quartz/xml/ValidationException.java:
```diff
@@ -127,10 +126,6 @@ public String getMessage() {
         for (Exception e : getValidationExceptions()) {
-            if (!first) {
-                sb.append('\n');
-                first = false;
```
This looks to be a bug where this line belongs in an else.

jhouserizer replied:
Agreed, it does look like a bug

bcode2 commented on quartz/src/main/java/org/quartz/core/jmx/JobDataMapSupport.java:
```diff
@@ -79,9 +78,7 @@ public static CompositeData toCompositeData(String key, String value) {
-        Iterator<String> iter = jobDataMap.keySet().iterator();
-        while (iter.hasNext()) {
-            String key = iter.next();
+        for (String key : jobDataMap.keySet()) {
```
yes, you right even my beloved  Sonarqube is complaining. I was  fixing simple issues first before going to the complex ones  :)  and grouping them in categories. Will fix them.... :)

bcode2 commented on quartz/src/main/java/org/quartz/impl/jdbcjobstore/StdJDBCDelegate.java:
```diff
@@ -3077,11 +3070,10 @@ protected ByteArrayOutputStream serializeJobData(JobDataMap data)
-            Map.Entry<?, ?> entry = (Map.Entry<?, ?>) value;
 
-            try (ByteArrayOutputStream baos = serializeObject(entry.getValue())) {
+            try (ByteArrayOutputStream baos = serializeObject(((Map.Entry<?, ?>) value).getValue())) {
```
umm I would prefer try-with-resources. Much safer

chrisdennis replied:
It's not safer - they're exactly the same in terms of safety. If `serializeObject(...)` throws there is nothing to close... and if it doesn't then we close the returned reference. Either way anything created is closed.

bcode2 commented on quartz/src/main/java/org/quartz/impl/jdbcjobstore/JobStoreSupport.java:
```diff
@@ -1274,33 +1270,29 @@ public boolean removeJobs(final List<JobKey> jobKeys) throws JobPersistenceExcep
 
-                        return allFound ? Boolean.TRUE : Boolean.FALSE;
-                    }
+                    return allFound ? Boolean.TRUE : Boolean.FALSE;
```
but with the original  code or current solution  at the end we need to cast  or it wont compile:(

`public boolean removeTriggers(final List<TriggerKey> triggerKeys)
            throws JobPersistenceException {
        return (boolean) executeInLock(
                LOCK_TRIGGER_ACCESS,
                (TransactionCallback) conn -> {
                    boolean allFound = true;

                    // FUTURE_TODO: make this more efficient with a true bulk operation...
                    for (TriggerKey triggerKey : triggerKeys)
                        allFound = removeTrigger(conn, triggerKey) && allFound;

                    return Boolean.valueOf(allFound);
                });
    }`

chrisdennis replied:
Weird, the above compiles for me, albeit with a warning about an unchecked assignment. More importantly though, if I remove the raw type, and the cast it all works cleanly (no compile errors or warnings):
```java
public boolean removeTriggers(final List<TriggerKey> triggerKeys)
        throws JobPersistenceException {
    return executeInLock(
            LOCK_TRIGGER_ACCESS,
            conn -> {
                boolean allFound = true;

                // FUTURE_TODO: make this more efficient with a true bulk operation...
                for (TriggerKey triggerKey : triggerKeys)
                    allFound &= removeTrigger(conn, triggerKey);

                return allFound;
            });
}
```

jhouserizer commented:
Can you help me understand where this all stands at this point?

bcode2 commented:
> Can you help me understand where this all stands at this point?

Hi @jhouserizer  give me a second . I guess I have already addressed all comments. Will rollback one  of those that are still unfixed to allow you to merge it. In another smaller PR will try to fix those referenced  by @chrisdennis  and @melloware

bcode2 commented:
Well @jhouserizer , @chrisdennis I guess  everything has been solved
